### DISCUSSION TITLE
Do both the .glb and .usdz files get loaded each time the model is rendered?

### DISCUSSION THREAD

**Discussion by roman-ocuon** [google/model-viewer#1914] [Q&A]
Currently I am setting both the `src `and `iosSc ` assets for each model, so that it can be viewed on all compatible devices. 
I'm trying gauge storage costs at scale and was wondering how these assets get loaded. Does model-viewer always load both? Does it conditionally load based on device/browser? Guess the overarching question is : what constitutes the `load` event and does this differ for iOS/Android.

**Comment by elalish** [comment]
The glb is always loaded as this is what is used to render within model-viewer itself. The `ios-src` URI is passed to QuickLook when AR is activated on an iOS device and QuickLook then downloads it, which explains the lag in getting the AR experience open, and why WebXR is so much faster (no re-download). Even on Android, if you use SceneViewer it must also redownload the glb, as we can't pass more than a URI to the intent. Unfortunately we don't have a choice until webKit implements WebXR.

**Comment by roman-ocuon** [comment]
Thank you for the quick response. Just to confirm, in terms of network usage, at the point in time where the AR experience is activated, the overall size of the assets required for the experience is 2x the size of the .glb file for Android, and .glb + .usdz sizes on iOS. And as a followup, in the case of Android, would the glb get re-downloaded even if the browser cached the initial glb file?

**Comment by elalish** [comment]
The key is to remember that when activating the quick-look or scene-viewer AR modes, you are leaving the browser entirely for a native app, so they aren't even in the same process and there is definitely no shared memory. When activating the webxr version of AR (currently Android-only), you stay in the browser and nothing is re-downloaded, in fact it keeps using the same GL context so there is no extra setup at all.
